processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1024,17 +1027,20 @@ public QueryableIndex load(File inDir, ObjectMapper mapper) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
-      Map<String, Object> metadata = null;
+      Metadata metadata = null;
       ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
       if (metadataBB != null) {
         try {
           metadata = mapper.readValue(
               serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
-              new TypeReference<Map<String, Object>>()
-              {
-              }
+              Metadata.class
```

Comment:
what do we currently put in metadata.drd? Will the values that we currently write for old segments be accessible somehow?